tests/test_codegen_advanced.cpp

```
@@ -251,6 +251,87 @@ TEST_F(CodeGenTest, LambdaArgTypeError) {
+}
+
+TEST_F(CodeGenTest, OuterReassignAfterCaptureOK) {
+    std::string src =
+        "base = 10\n"
+        "add_base = (x: int) => x + base\n"
+        "base = 99\n"
+        "print(add_base(5))";
+    EXPECT_EQ(runSource(src), "15\n");
+}
+
+TEST_F(CodeGenTest, CapturedRecordFieldAssignOK) {
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)